src/components/reportPage/UploadBox.jsx

```diff
@@ -2,138 +2,44 @@ import React, { useCallback, useEffect, useState } from "react";
-        } else if (errorType === "duplicate-file") {
-          errorFiles.duplicateFiles = [...errorFiles.duplicateFiles, file.name];
-        } else if (errorType === "file-too-large") {
-          errorFiles.overMaxSize = [...errorFiles.overMaxSize, file.name];
-        }
+    const rejectFiles = fileRejections.length;
+    if (rejectFiles) {
+      if (rejectFiles > fileLimit) {
+        setAlert(MAX_FILE_ALERT);
+      } else if (rejectFiles < fileLimit) {
+        // To display the duplicate files uploaded
+        const duplicateFileAlert = fileRejections.map((f) => " " + f.file.name);
+        setAlert("You have already uploaded:" + duplicateFileAlert);
+      } else {
+        setAlert(INVALID_FILETYPE);
       }
-
-      const msg = createErrorMsg(errorList);
-      setAlert(msg);
     }
   };
 
-  // @param: errorObj should object composed of two objs: errorFiles, errorMsgs, both have identical property names
-  const createErrorMsg = (errorObj) => {
-    const { errorFiles, errorMsgs } = errorObj;
-    const errorTypes = Object.keys(errorFiles);
-
-    // loops through error types for file arrays that aren't empty and adds to rejected files
-    const errors = errorTypes.map((type) =>
-      errorFiles[type].length ? (
-        <>
-          <p style={{ marginBottom: "0.2rem" }}>{errorMsgs[type]}</p>{" "}
-          <ul>
-            {" "}
-            {errorFiles[type].map((file) => (
-              <li key={file}>{file}</li>
-            ))}
-          </ul>{" "}
-        </>
-      ) : null
-    );
-
-    // CSS for errors and errorHeading
-    const errorStyle = {
-      marginLeft: "1rem",
-      marginRight: "1.5rem",
-      overflowWrap: "break-word",
-      textAlign: "left",
-    };
-    const errorHeadingStyle = { marginBottom: "0.5rem", textAlign: "left" };
-
-    // final message jsx
-    const msg = (
-      <>
-        <p style={errorHeadingStyle}>
-          {" "}
-          <b> Rejected Files: </b>{" "}
-        </p>
-        <div style={errorStyle}>{errors}</div>
-      </>
-    );
-    return msg;
-  };
-
-  // Customer file validation function
+  // Cusomter file validation function
   const validateFiles = (file) => {
-    // Tests if can accept more files
-    if (currFiles.length == fileLimit) {
-      return { code: "over-max-filelimit" };
+    if (!props.currFiles.every((currFile) => currFile.name != file.name)) {
+      return {};
     }
-
-    // Tests duplicate files
-    else if (!props.currFiles.every((currFile) => currFile.name != file.name)) {
-      return { code: "duplicate-file" };
-    }
-
     // Returns null if files should be accepeted
     return null;
   };
-
   const acceptedFileTypes = ["image/jpeg", "image/png", "video/mp4"];
-  const maxFiles = fileLimit ? fileLimit : 0; // 0 unlimited files
-  const maxSize = maxFileSize ? maxFileSize * 1000 * 1000 : Infinity; // 0 is unlimited file size, in bin bytes
+  const maxFiles = props.fileLimit == Infinity ? 0 : props.fileLimit;
```

Comment:
This is Ok, but I would rather have an upper limit, having `Infinity` might break the website in ways we do not anticipate.